tests/test_utils.py

```diff
@@ -124,6 +136,38 @@ def test_ingest_injection_catalog(self):
         injected_catalog = self.writeable_butler.get(input_dataset_refs[0])
         self.assertTrue(all(self.injection_catalog == injected_catalog))
 
+    def test_consolidate_injected_catalogs(self):
+        self.injected_catalog.add_column(col=0, name="injection_flag")
+        catalog_dict = {"g": self.injected_catalog, "r": self.injected_catalog}
+        output_catalog = consolidate_injected_deepCoadd_catalogs(
+            catalog_dict=catalog_dict,
+            skymap=self.skyMap,
+            tract=9,
+            pixel_match_radius=0.1,
+            get_catalogs_from_butler=False,
+            col_ra="ra",
+            col_dec="dec",
+            col_mag="mag",
+            isPatchInnerKey="injected_isPatchInner",
+            isTractInnerKey="injected_isTractInner",
+            isPrimaryKey="injected_isPrimary",
+            injectionKey="injection_flag",
+        )
+        self.assertEqual(len(output_catalog), 30)
+        expected_columns = {
+            "injected_id",
+            "ra",
+            "dec",
+            "source_type",
+            "g_mag",
+            "r_mag",
+            "injection_flag",
+            "injected_isPatchInner",
+            "injected_isTractInner",
+            "injected_isPrimary",
+        }
+        self.assertEqual(set(output_catalog.columns), expected_columns)
+
```

Comment:
You're not currently testing any of the flagging capabilities, which I think is an important aspect of `consolidate_injected_deepCoadd_catalogs`.

Perhaps, in `setUp`, you can modify a few rows to set patch/tract and injection flags, and then here, add an assert for the number of `injected_isPrimary` rows that _should_ be flagged.